src/Raven.Server/ServerWide/SecretProtection.cs

```diff
@@ -195,5 +222,245 @@ public static byte[] Unprotect(byte[] secret, byte[] entropy)
             }
             return unprotectedData;
         }
+
+        public RavenServer.CertificateHolder LoadCerificateWithExecutable()
+        {
+            var path = _config.CertificateExec;
+            var args = _config.CertificateExecArguments;
+            var timeout = _config.CertificateExecTimout;
+
+            if (string.IsNullOrEmpty(path))
+                return null;
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = path,
+                    Arguments = args,
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    CreateNoWindow = false
+                }
+            };
+
+            var sw = Stopwatch.StartNew();
+
+            try
+            {
+                process.Start();
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"Unable to get certificate by executing {path} {args}. Failed to start process.", e);
+            }
+
+            var ms = new MemoryStream();
+            var readErrors = process.StandardError.ReadToEndAsync();
+            var readStdOut = process.StandardOutput.BaseStream.CopyToAsync(ms);
+
+            if (process.WaitForExit(_config.CertificateExecTimout) == false)
+            {
+                throw new InvalidOperationException($"Unable to get certificate by executing {path} {args}, waited for {timeout} ms but the process didn't exit.");
```

Comment:
Need to include all the standard error that we read so far here.
Need to kill the process.